### PR TITLE
Fix exposure of daysInMonth, and make use of it in the tests.

### PR DESCRIPTION
**Pull request by zenhack**
See the commit messages.

### CODE REVIEW

File: tests/Tests.hs [hunk 1]
```diff
@@ -105,9 +102,10 @@ instance Arbitrary Month where
 instance Arbitrary DateTime where
     arbitrary = DateTime <$> arbitrary <*> arbitrary
 instance Arbitrary Date where
-    arbitrary = Date <$> choose dateRange
-                     <*> arbitrary
-                     <*> choose (1,28)
+    arbitrary = do
+        year <- choose dateRange
+        month <- arbitrary
+        Date year month <$> choose (1, daysInMonth year month)
```
Comment (NicolasDP) [inline tests/Tests.hs]: Why not keeping using Applicative?

Comment (zenhack) [inline tests/Tests.hs]: This introduces a data dependency of the day on the other two fields, so you can't just apply the constructor to all three like before. I did't see an obvious way of doing this with Applicative that was reasonably clear, did I miss something obvious?

Comment (vincenthz) [inline tests/Tests.hs]: I think Nicolas just missed the dependency; it's indeed the right thing to do here

Comment (NicolasDP) [inline tests/Tests.hs]: oops. sorry

Comment (NicolasDP) [inline tests/Tests.hs]: It's me how missed something obvious.

**Comment by NicolasDP** [comment]
Looking good, thanks.

**Pull request merged**